contrib/dockerd-rootless-setuptool.sh

```diff
@@ -143,7 +143,11 @@ init() {
 
 	# instruction: iptables dependency check
 	faced_iptables_error=""
-	if ! command -v iptables > /dev/null 2>&1 && [ ! -f /sbin/iptables ] && [ ! -f /usr/sbin/iptables ]; then
+	iptables_command=$(PATH=$PATH:/sbin:/usr/sbin command -v iptables 2> /dev/null) || :
+	iptables_version=""
+	if [ -n "$iptables_command" ]; then
+		iptables_version=$($iptables_command --version 2> /dev/null) || :
```

Comment:
Actually; so `iptables_version` is only used to set `iptables_module`, correct? In that case, we could just put all here? Or won't that work?

```suggestion
	# Many OSs now use iptables-nft by default so, check for module nf_tables by default. But,
	# if "iptables --version" worked and reported "legacy", check for module ip_tables instead.
	iptables_module="nf_tables"
	iptables_command=$(PATH=$PATH:/sbin:/usr/sbin command -v iptables 2> /dev/null) || :
	if [ -n "$iptables_command" ]; then
		iptables_version=$($iptables_command --version 2> /dev/null) || :
		case $iptables_version in
			*legacy*) iptables_module="ip_tables" ;;
		esac
```

Reply:
The first block (`instruction: iptables dependency check`) is working out whether there's an `iptables` command, and the second block (`instruction: ip_tables module dependency check`) works out whether the kernel module is loaded.

So, I thought it was best for the second block to use the command the first block found - rather than finding the command again, but maybe a bit differently.

If that makes sense (?) ... I'll swap the grep for case, as you suggest above.

Reply:
I've taken option 1 ... can update again if needed!

Reply:
Sorry - misread, this is the first block ... done now.